### DISCUSSION TITLE
Permit access to user from trusted domain

### DISCUSSION THREAD

**Discussion by x09** [neutrinolabs/xrdp#3169] [Q&A] [answered]
Hi,

I have two AD domain (trust)

```
[root@alt ~]# sssctl domain-list
TEST.ALT
win.test
```

I have users in "rdp_users" group from trusted domain

```
[root@alt ~]# id  winuser1@WIN.TEST
uid=679001103(winuser1@win.test) gid=679001103(winuser1@win.test) группы=679001103(winuser1@win.test),679001106(rdp_users@win.test),679000513(domainusers@win.test)
```

or

```
[root@alt ~]# id WIN.TEST\\winuser1
uid=679001103(winuser1@win.test) gid=679001103(winuser1@win.test) группы=679001103(winuser1@win.test),679001106(rdp_users@win.test),679000513(domainusers@win.test)
```
group memebers work fine also

```
[root@alt ~]# getent group rdp_users@WIN.TEST
rdp_users@win.test:*:679001106:winuser1@win.test,winuser2@win.test

```
or

```
[root@alt ~]# getent group WIN.TEST\\rdp_users
rdp_users@win.test:*:679001106:winuser1@win.test,winuser2@win.test
```

How i can specified  TerminalServerUsers option in /etc/xrdp/sesman.ini for allow this group (rdp_users@win.test)?

I tried
TerminalServerUsers=rdp_users@WIN.TEST
TerminalServerUsers=rdp_users@win.test
TerminalServerUsers=WIN.TEST\\rdp_users
TerminalServerUsers=WIN.TEST\\rdp_users
TerminalServerUsers=WIN.TEST\rdp_users
TerminalServerUsers=WIN.TEST\rdp_users

but always i get login denied for user WIN.TEST\winuser1 (or winuser1@WIN.TEST)

```
Jul 19 11:37:40 alt xrdp-sesman[4097]: pam_sss(xrdp-sesman:auth): authentication success; logname= uid=0 euid=0 tty=xrdp-sesman ruser= rhost= user=WIN.TEST\winuser1
Jul 19 11:37:40 alt audit[4097]: USER_AUTH pid=4097 uid=0 auid=4294967295 ses=4294967295 msg='op=PAM:authentication grantors=pam_sss acct="WIN.TEST\winuser1" exe="/usr/sbin/xrdp-sesman" hostname=? addr=? terminal=xrdp-sesman res=success'
Jul 19 11:37:40 alt audit[4097]: USER_ACCT pid=4097 uid=0 auid=4294967295 ses=4294967295 msg='op=PAM:accounting grantors=pam_sss acct="WIN.TEST\winuser1" exe="/usr/sbin/xrdp-sesman" hostname=? addr=? terminal=xrdp-sesman res=success'
Jul 19 11:37:40 alt xrdp-sesman[4097]: (4097)(140285919799104)[INFO ] login denied for user WIN.TEST\winuser1
Jul 19 11:37:40 alt xrdp[4102]: (4102)(140067377204544)[INFO ] xrdp_wm_log_msg: login failed for display 0
```

which format of trusted domain for  TerminalServerUsers ?

**Comment by matt335672** [comment]
We added a patch to make this work better to v0.9.24 ([release notes](https://github.com/neutrinolabs/xrdp/releases/tag/v0.9.24)). Previous versions suffer from #2806.

Which version of xrdp are you running?

**Reply by x09** [reply, marked as answer]
> We added a patch to make this work better to v0.9.24 ([release notes](https://github.com/neutrinolabs/xrdp/releases/tag/v0.9.24)). Previous versions suffer from #2806.
> 
> Which version of xrdp are you running?

Update to 0.10
set TerminalServerUsers=rdp_users@win.test
Work fine! Thnks!

**Comment by x09** [comment]
Additionally.
I can not  add more than one group in TerminalServerUsers ?
Like as:  `TerminalServerUsers=rdp_users@domain1, rdp_users@domain2 ?`

**Reply by matt335672** [reply]
No.

Use another group for that.
